build.gradle

```diff
@@ -98,29 +98,61 @@ spotbugs {
     excludeFilter = new File(rootDir, "config/metrics/findbugs/findbugs-exclude.xml")
 }
 
-//publishing {
-//    publications {
-//        mavenJava(MavenPublication) {
-//            
-//            from components.java
-//
-//            artifactId 'splash-screen'   // for published jar files
-//			
-//            artifact sourceJar {
-//                classifier "sources"
-//            }
-//            artifact javadocJar {
-//                classifier "javadoc"
-//            }
-//        }
-//    }
-//}
-
-//artifactory {
-//    publish {
-//        defaults {
-//            publications('mavenJava')
-//        }
-//    }
-//}
+publishing {
+    publications {
+        "$project.name"(MavenPublication) {
+            // Without this we get a .pom with no dependencies
+            from components.java
+
+            artifact source: sourceJar, classifier: 'sources'
+            artifact source: javadocJar, classifier: 'javadoc'
+
+            repositories {
+                maven {
+                    name = 'TerasologyOrg'
+
+                    if (rootProject.hasProperty("publishRepo")) {
+                        // This first option is good for local testing, you can set a full explicit target repo in gradle.properties
+                        url = "http://artifactory.terasology.org/artifactory/$publishRepo"
+                        allowInsecureProtocol true // 😱
+                        logger.info("Changing PUBLISH repoKey set via Gradle property to {}", publishRepo)
+                    } else {
+                        // Support override from the environment to use a different target publish org
+                        String deducedPublishRepo = System.getenv()["PUBLISH_ORG"]
```

Comment:
Should we prefix this env variable with something more specific, maybe `MAVEN_PUBLISH_ORG` or even `TERASOLOGY_MVN_PUBLISH_ORG`? Reducing the chances that there is some incidental overlap.

Reply:
We could, and I'm not opposed or anything, but in that case we should probably make that change everywhere that uses this setup, which is multiple repos now. Then we should also update the env var set in Jenkins that varies this, which then also means we should update the Jenkins setup doc, and so on :-)

For now I'd like to avoid that extra effort for a naming improvement - and long term I'm hoping we can just wrap this in a Gradle plugin or something so we can avoid the duplication everywhere and just change it once.

(That said, we could mitigate by just setting _two_ env vars in Jenkins, both old and new but ...)